Core/TransactionHandler.php

```diff
@@ -0,0 +1,231 @@
+            $oCurrency = Registry::getConfig()->getCurrencyObject($sCurrencyName);
+
+            $oTransaction->setAmount(new Amount(
+                Registry::getUtils()->fRound($fAmount, $oCurrency),
+                $sCurrencyName
+            ));
+        }
+
+        // use the backend service to process the action
+        $oBackendService = new BackendService($oConfig, $this->_oLogger);
+        $oResponse = $oBackendService->process($oTransaction, $sActionTitle);
+        $this->_oLogger->debug('response from ' . $sActionTitle . ' action: ' . $oResponse->getRawData());
+
+        return $this->_handleProcessActionResponse($oResponse, $oOrder, $oBackendService);
+    }
+
+    /**
+     * Handles the success and error response from the post-processing action.
+     *
+     * @param Response       $oResponse
+     * @param Order          $oOrder
+     * @param BackendService $oBackendService
+     *
+     * @return array either [status => success] or [status => error, message => errorMessage]
+     */
+    private function _handleProcessActionResponse($oResponse, $oOrder, $oBackendService)
+    {
+        if ($oResponse instanceof FailureResponse) {
+            $sErrors = '';
+
+            foreach ($oResponse->getStatusCollection()->getIterator() as $oItem) {
+                $sErrors .= $oItem->getDescription() . "<br>";
+                $this->_oLogger->error($oItem->getCode() . ': ' . $oItem->getDescription());
+            }
+
+            return $this->_getErrorMessage(Helper::translate('error_transaction') . '<br>' . $sErrors);
+        }
+
+        if ($oResponse instanceof SuccessResponse) {
+            // create new transaction entry in the database
+            $this->_handleNotificationSuccess($oResponse);
+
+            // also update the order state in the database
+            $sUpdatedOrderState = $oBackendService->getOrderState($oResponse->getTransactionType());
+            $oOrder->oxorder__wdoxidee_orderstate = new Field($sUpdatedOrderState);
+            $oOrder->save();
+        }
+
+        return $this->_getSuccessMessage();
+    }
+
+    /**
+     * Returns the appropriate transaction object for the desired operation.
+     *
+     * @param Payment_Method $oPayment
+     * @param string         $sActionTitle
+     *
+     * @return Transaction
+     */
+    private function _getTransactionObject($oPayment, $sActionTitle)
+    {
+        switch ($sActionTitle) {
+            case Operation::CANCEL:
+                return $oPayment->getCancelTransaction();
+            case Operation::PAY:
+                return $oPayment->getCaptureTransaction();
+        }
+
+        return null;
+    }
+
+    /**
+     * Handles the success response from the post-processing operation.
+     * A new transaction entry is added to the database.
+     *
+     * @param SuccessResponse $oResponse
+     *
+     * @return null
+     */
+    private function _handleNotificationSuccess($oResponse)
```

Comment:
is his the response handling or noification handling of a transaction. The phpdoc is saying its response but the method name that its a notification, also notifications are usually handled the same as the initial ones so no need for new logic.